site/class-3-clean/vincentkempers/index.html

```diff
@@ -0,0 +1,231 @@
+var height = svg.attr('height') - margin.top - margin.bottom;
+var g = svg.append('g').attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+var x = d3.scaleTime().range([0, width]);
+var y = d3.scaleLinear().range([height, 0]);
+var z = d3.scaleOrdinal(['#feca2f', '#2ffe63', '#2f63fe', '#fe2fca']);
+
+var knmiCodes = {
+  209: 'IJmond',
+  210: 'Valkenburg',
+  215: 'Voorschoten',
+  225: 'IJmuiden',
+  235: 'De Kooy',
+  240: 'Schiphol',
+  242: 'Vlieland',
+  248: 'Wijdenes',
+  249: 'Berkhout',
+  251: 'Hoorn (Terschelling)',
+  257: 'Wijk aan Zee',
+  258: 'Houtribdijk',
+  260: 'De Bilt',
+  265: 'Soesterberg',
+  267: 'Stavoren',
+  269: 'Lelystad',
+  270: 'Leeuwarden',
+  273: 'Marknesse',
+  275: 'Deelen',
+  277: 'Lauwersoog',
+  278: 'Heino',
+  279: 'Hoogeveen',
+  280: 'Eelde',
+  283: 'Hupsel',
+  285: 'Huibertgat',
+  286: 'Nieuw Beerta',
+  290: 'Twenthe',
+  308: 'Cadzand',
+  310: 'Vlissingen',
+  311: 'Hoofdplaat',
+  312: 'Oosterschelde',
+  313: 'Vlakte v.d. Raan',
+  315: 'Hansweert',
+  316: 'Schaar',
+  319: 'Westdorpe',
+  323: 'Wilhelminadorp',
+  324: 'Stavenisse',
+  330: 'Hoek van Holland',
+  331: 'Tholen',
+  340: 'Woensdrecht',
+  343: 'R’dam-Geulhaven',
+  344: 'Rotterdam',
+  348: 'Cabauw',
+  350: 'Gilze-Rijen',
+  356: 'Herwijnen',
+  370: 'Eindhoven',
+  375: 'Volkel',
+  377: 'Ell',
+  380: 'Maastricht',
+  391: 'Arcen'
+};
+
+// You may start editing stuff here (but don’t change the file-name).
+d3.text('index.txt')
+  .get(onload);
+
+function onload (err, doc) {
+  if (err) throw err;
+
+  // Tijd moet isco format hebben "2017-08-31T23:00:00.000Z"
+
+  // Use this place to transform doc (without changing it manually) into
+  // something that looks like `example.json`:
+  // https://github.com/cmda-fe3/course-17-18/blob/master/site/class-3/clean/example.json
+  //
+  // Notes:
+  //
+  // 1. Use the `knmiCodes` variable to transform from KNMI’s codes to place
+  //    names
+  // 2. Use actual Date objects instead of strings for all `date` keys:
+  //    https://developer.mozilla.org/JavaScript/Reference/Global_Objects/Date
+  //
+  // Store the results in the variable `places`:
+  // variabelen
+
+  var deleteHeader = doc.indexOf('STN,YYYYMMDD');
+  doc = doc.slice(deleteHeader);
+  var end = doc.indexOf('\n', '240');
+  doc = doc.slice(end).trim();
+  doc = doc.replace(/ +/g, '');
+  var parseTime = d3.timeParse('%Y%m%dT%H');
+  var testData = d3.csvParseRows(doc);
+  var places = d3.csvParseRows(doc, map);
+
+  places = d3.nest()
+    .key(function(d) { 
+      return d.name; 
+    })
+    .entries(places).map(function (aSort) {
+      return {
+        name: aSort.key,
+        values: aSort.values
+      };
+    });
+
+  function map(placeData) {
+    return {
+      name: knmiCodes[placeData[0]],
+      date: parseTime(placeData[1].concat('T', placeData[2])),
+      temperature: placeData[7]/10
+    }
+  }
+
+ places.pop();
```

Comment:
Now you’re always removing the last entry, which could be Huibertgat, or could be something else!
This works for this dataset, but wouldn’t work if you’d have different data.

A much better approach would be to filter out missing values, and only include things in `places` if they existed.